org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -222,6 +222,11 @@ public void fileChanged(String uriString, CHANGE_TYPE changeType) {
 			Optional<IBuildSupport> bs = getBuildSupport(resource.getProject());
 			if (bs.isPresent()) {
 				IBuildSupport buildSupport = bs.get();
+
+				if (JDTUtils.isExcludedFile(buildSupport.getExcludedFilePatterns(), uriString)) {
```

Comment:
Since this is still dealing with the `uriString`, how about put it before `IResource resource = JDTUtils.getFileOrFolder(uriString);`

Reply:
I am using `resource` to get `buildSupport`. Therefore, if we check `isExcludedFile` before fetching resources, some manipulation of the `uriString` will be required. It will be a doubling of effort.

Reply:
OK, fair.